streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -674,6 +676,8 @@ class StreamingContext private[streaming] (
           logWarning("StreamingContext has already been stopped")
         case ACTIVE =>
           scheduler.stop(stopGracefully)
+          // De-registering Streaming Metrics of the StreamingContext
+          env.metricsSystem.removeSource(streamingSource)
```

Comment:
I think it is OK for normal case, but what if an exception is met after metrics is successfully registered, but before changing the state into `ACTIVE`, according to the code, it will try the exception and change state into `STOPPED`, so at that situation, if we call stop(), we will never de-register the metrics  source according to the current implementation.